File: 06-Flutter/fake_store_app/lib/models/product.dart
class Product {
  final int id;
  final String title;
  final double price; // 19.99 처럼 서양에서는 소숫점을 자주 쓰므로 double type을 사용했음
  final String description;
  final String image;
  final Map<String, dynamic> rating;

  Product({
    required this.id,
    required this.title,
    required this.price,
    required this.description,
    required this.image,
    required this.rating,
  });

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'price': price,
      'description': description,
      'image': image,
      'rating': rating,
    };
  }

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      title: json['title'],
      price: (json['price'] as num).toDouble(),
      description: json['description'],
      image: json['image'],
      rating: json['rating'],
    );
  }
}

File: 11-Tuist-TCA/HelloTuist/Modules/Network/Sources/Models/Product.swift
import Foundation

public struct Product: Decodable, Identifiable, Equatable {
    public let id: Int
    public let title: String
    public let price: Double
    public let description: String
    public let category: String
    public let image: String // 이미지 URL 문자열
    public let rating: Rating
    
    public struct Rating: Decodable, Equatable {
        public let rate: Double
        public let count: Int
    }
    
    // Equatable 준수를 위한 구현 ( 테스트 코드 등에서 비교를 위해 필요할 수 있음)
    public static func == (lhs: Product, rhs: Product) -> Bool {
        return lhs.id == rhs.id
    }
}
